bitcoin/src/taproot.rs

```diff
@@ -111,18 +115,23 @@ impl From<&ScriptLeaf> for TapLeafHash {
     fn from(leaf: &ScriptLeaf) -> TapLeafHash { leaf.leaf_hash() }
 }
 
-impl TapBranchHash {
+impl TapNodeHash {
     /// Computes branch hash given two hashes of the nodes underneath it.
-    pub fn from_node_hashes(a: sha256::Hash, b: sha256::Hash) -> TapBranchHash {
-        let mut eng = TapBranchHash::engine();
+    pub fn from_node_hashes(a: TapNodeHash, b: TapNodeHash) -> TapNodeHash {
+        let mut eng = TapNodeHash::engine();
         if a < b {
             eng.input(&a);
             eng.input(&b);
         } else {
             eng.input(&b);
             eng.input(&a);
         };
-        TapBranchHash::from_engine(eng)
+        TapNodeHash::from_engine(eng)
+    }
+
+    /// Computes the [`TapNodeHash`] from a script and a leaf version.
+    pub fn from_script(script: &Script, ver: LeafVersion) -> TapNodeHash {
+        TapNodeHash::from_inner(TapLeafHash::from_script(script, ver).into_inner())
```

Comment:
In e03b9457111848cb26f3764b2d9cdc9d9ea21534:

Can you add a comment here highlighting that you're actually casting between two hash types here, since it's maybe hard to see in the `from_inner(X.into_inner())` form, explaining that this is because *leaves* of a taptree use a different hash prefix than the *internal nodes*, but the Merkle tree logic does not care about the distinction. Maybe also link to the BIP and/or to #1393.

I think a regular comment is fine, no need to doccomment this, since it's an internal implementation detail.

Reply:
Also, I agree with Kix's suggestion below that we add a conversion method from `TapLeafHash` to `TabBranchHash`, so this line can be simplified to `TapLeafHash::from_script(script, ver).into()`.

Reply:
Adressed

Reply:
Since there's `From` now we might as well use it here?